CONFIGURATION.md

```diff
@@ -97,6 +97,8 @@ rebalance_cb                             |  C  |                 |
 offset_commit_cb                         |  C  |                 |               | Offset commit result propagation callback. (set with rd_kafka_conf_set_offset_commit_cb()) <br>*Type: pointer*
 enable.partition.eof                     |  C  | true, false     |          true | Emit RD_KAFKA_RESP_ERR__PARTITION_EOF event whenever the consumer reaches the end of a partition. <br>*Type: boolean*
 check.crcs                               |  C  | true, false     |         false | Verify CRC32 of consumed messages, ensuring no on-the-wire or on-disk corruption to the messages occurred. This check comes at slightly increased CPU usage. <br>*Type: boolean*
+enable.idempotence                       |  P  | true, false     |         false | When set to `true`, the producer will ensure that messages are successfully produced exactly once and in the original produce order. The following configuration properties are adjusted automatically (if necessary) when idempotence is enabled: `max.inflight.requests.per.connection` is capped at 5, `retries` must be greater than 0, `acks` is  set to `all`, `queuing.strategy` is set to `fifo`. <br>*Type: boolean*
```

Comment:
it would be good to clarify here whether these are automatically adjusted only in the case they haven't been explicitly set or whether setting this overwrites explicit settings as well (which is how it reads to me). in the latter case, I think the config should be considered invalid and call to `rd_kafka_new` fail (haven't looked at what is actually done yet).

Reply:
It works the way you describe it, kafka_new() will fail if a user has set incompatible properties. But yeah, it needs to be made clear here.

Unmodified properties are automatically adjusted.

Reply:
sounds good